src/test/java/com/acme/conferencesystem/voting/http/VoteControllerIntegrationTest.java

```diff
@@ -0,0 +1,89 @@
+package com.acme.conferencesystem.voting.http;
+
+import com.acme.conferencesystem.AbstractIntegrationTest;
+import com.acme.conferencesystem.ContainerConfig;
+import com.acme.conferencesystem.cfp_proposals.business.Proposal;
+import com.acme.conferencesystem.users.business.User;
+import com.acme.conferencesystem.users.business.UserRole;
+import com.acme.conferencesystem.voting.business.Vote;
+import io.restassured.http.ContentType;
+import org.instancio.Instancio;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.modulith.test.ApplicationModuleTest;
+
+import static io.restassured.RestAssured.given;
+import static org.instancio.Select.field;
+
+@ApplicationModuleTest(
+        mode = ApplicationModuleTest.BootstrapMode.DIRECT_DEPENDENCIES,
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@Import(ContainerConfig.class)
+class VoteControllerIntegrationTest extends AbstractIntegrationTest {
+
+    @Test
+    void voteProposal() {
+        //todo
+        // 1. create an organizer
+        User user = Instancio.of(User.class)
+                .ignore(field(User::id))
+                .set(field(User::role), UserRole.ORGANIZER)
+                .create();
+
+        // Submit user
+        var newUserId = given(requestSpecification)
+                .contentType(ContentType.JSON)
+                .body(user)
+                .when()
+                .post("/users")
+                .then()
+                .statusCode(201)
+                .extract()
+                .path("id");
+
+        // 2. create a proposal
+        Proposal proposal = Instancio
+                .of(Proposal.class)
+                .set(field(Proposal::speakerId), newUserId)
```

Comment:
You'll have an error, newUserId is an Object and you need a UUID, otherwise, Instancio will not be able to create the object and it'll return null.

```java
  // Submit user
        String newUserId = given(requestSpecification)
                .contentType(ContentType.JSON)
                .body(user)
                .when()
                .post("/users")
                .then()
                .statusCode(201)
                .extract()
                .path("id");

        // 2. create a proposal
        Proposal proposal = Instancio.of(Proposal.class)
                .ignore(field(Proposal::id))
                .set(field(Proposal::speakerId), UUID.fromString(newUserId))
                .create();
```

Reply:
Thanks